pkg/rancher-desktop/backend/containerClient/nerdctlClient.ts

```diff
@@ -70,6 +70,35 @@ export class NerdctlClient implements ContainerEngineClient {
     }
   }
 
+  /**
+   * Like running this.vm.execCommand, but retries the command if no output
+   * is produced. Is a workaround for a strange behavior of this.vm.execCommand:
+   * sometimes nothing is returned from stdout, as though it did not run at
+   * all. See https://github.com/rancher-sandbox/rancher-desktop/issues/4473
+   * for more info.
+   */
+  protected async execCommandWithRetries(...command: string[]): Promise<void>;
+  protected async execCommandWithRetries(options: execOptions, ...command: string[]): Promise<void>;
+  protected async execCommandWithRetries(options: execOptions & { capture: true }, ...command: string[]): Promise<string>;
+  protected async execCommandWithRetries(optionsOrArg: execOptions | string, ...command: string[]): Promise<void | string> {
+    let options: execOptions & { capture?: boolean } = {};
+
+    if (typeof optionsOrArg === 'string') {
+      command = [optionsOrArg].concat(command);
+    } else {
+      options = optionsOrArg;
+    }
+
+    const maxRetries = 10;
+    let result: void | string;
+
+    for (let i = 0; i < maxRetries && !result; i++) {
+      result = await this.vm.execCommand(options, ...command);
```

Comment:
This loop only ever makes sense if `result` is a string (if it's a `void`, it'll always loop up to `maxRetries`).  It's probably to be explicit about it in the method signature (only take the third option).